### PR TITLE
Code cleanup / refactoring Goggles

### PR DESCRIPTION
**Pull request by commy2**
ref: #2144

### CODE REVIEW

**Comment by commy2** [comment]
- [ ] VR glasses

File: addons/goggles/functions/fnc_isDivingGoggles.sqf [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
-_result = [configFile >> "CfgGlasses" >> _glasses, configFile >> "CfgGlasses" >> "G_Diving"] call CBA_fnc_inheritsFrom;
-_result
+
+params ["_glasses"];
+
+local _config = configFile >> "CfgGlasses" >> _glasses;
+
+if (!isClass _config) exitWith {false};
+
+getNumber (_config >> "mode") == 1 // return
```
Comment (SilentSpike) [inline addons/goggles/functions/fnc_isDivingGoggles.sqf]: Out of curiosity, what does the `"mode"` property represent?

Comment (commy2) [inline addons/goggles/functions/fnc_isDivingGoggles.sqf]: `mode = 1;` is a config entry that changes the behaviour of goggles.
Usually goggles are hidden on the unit if it gets into the water even though they are equipped in that slot.
That is reversed when `mode` is set to 1. The goggles are only shown when in water. This is used for diving goggles in vanilla.

File: addons/goggles/functions/fnc_externalCamera.sqf [hunk 2]
```diff
@@ -6,15 +6,13 @@
...
+ * call ace_goggles_fnc_externalCamera;
  *
  * Public: Yes
  */
 #include "script_component.hpp"
 
-if (GVAR(showInThirdPerson)) exitWith { false };
-
-(cameraView in ["EXTERNAL", "GROUP"] || {call EFUNC(common,isFeatureCameraActive)})
+cameraView in ["EXTERNAL", "GROUP"] || EFUNC(common,isFeatureCameraActive) // return
```
Comment (SilentSpike) [inline addons/goggles/functions/fnc_externalCamera.sqf]: Isn't the `call` required here?

Comment (SilentSpike) [inline addons/goggles/functions/fnc_externalCamera.sqf]: Ah wait, I see, it's automatically lazy evaluation

Comment (commy2) [inline addons/goggles/functions/fnc_externalCamera.sqf]: The implemenation of lazy evaluation is nothing more than the addition of the alternative syntax of `&&` and `||` which is `BOOL && CODE` etc.

**Comment by commy2** [comment]
After some testing it seems that everything still works as expected. Merge?

**Pull request merged**